### PR TITLE
unit status trigger fixes

### LINKED ISSUE #2015: Conditions not run
## Describe the bug
Tested on classic with a warlock level 1

Import this
```
!TEvBVTTnq4)lfyaTOiowYoPO7ddij1DnTEYPMoTDByYHwIwInYKAIuo15d(3(UJK6fBNyNmu0IerYJ39CVWN7sOx4KqcpmWR3PEV9nHeviXVJ3PDo9i)3EuYBN9g)z0tdjlcjX4)bHxnA(Cfthg0nKqfrPYIRKCHoKCXGGjdghsIKYSy5DcYD8Cw45nBmioXSMgP5sHcvMstl04hXYPrLkTCr4mZIfmLIcIdRC7tYlaZ8YxOsL39IxfcgKl47Cxy75W(Qu4anUsxWtsyfO1cmoRBdZNRaaIyqxc(TQCgBjtOjLZNZ)riz6fNrMmLm5SXtAo8QcM5qYvdgoemDzbfDMqIxiPuXMQYzzzOtIlkraIFJ)MKNrxb2LiOGZzGhXi9LXUv4v4ILayn(TXEqyLQWZeUL0sT0DrafPuiWU(MZLzWrXSza0NyCQpmy4vV)6Hg1kACzmKe4VvyaSBaGjtQb8Nk8bHRAdmb2oIHAdYJJezRqHDi6dmAMo9rdrB68lO6OuMIajXrqutz(9FWvkUi5aPaB2crO)t3x7T3u(wr1nYMvHFPiMBRxFmpC)foBKIBL73Oab(MMXxY2TA5P6O9))xB)qU91aiwFZfP0c41kRGR08ivf2GhTjadaPaY8e0)mLpBvnw)AOQ2ABh2QLXS)TKxWITE(Mf5psbfgS0AA0T0zzgn3yMapVtpT72bOnHTjCDYZPUiNvebltDf6DFYL(7Wbuf52B(aV2MM88TWWujSKQLq9XVTz(7GEojMR(EPaOGxIVMeRQiyNyV0qzcpcirnIifV0PRxT(JRVX(VcMUSqS(g3j)T3)S(JmrSLxFjZPN)qgdPJJ86AzQHAibn7la3gc0aFONYTmw(zqQlspgHps5rf8fwxPDVbBUrifSgFDQlHXIG3NkS5atzcSz8yMwMJwDbLlEI3VDBJNL9MtJzU4ktbfpG0gY78czciIcQiI4wFgkBww7)MsGXSexJqBRPzYIywXuf)Ei057xVH01XTF9orYmzH9EGEb1H)SN5N9Xn11sUeSU7zIBhg2gMCEg)(7PfXa3o0EwZZRFduoZkytRc2p0tvMh3FRvZ)AXXZDsz)8xYDlBdtpdm9mW0RbMg5MlXxtVVGF)6B(CjngIIuayBA7loSp3s6)SnsnBZYM7Mu5SRNmcyVKWRD2edIdgfmOfwS0TJUEYWlR3)bN1XU5o267Wlk(8vnYvRzsDU1E1VT9vRYxZSvvPmEskYxbNNjPXU(1wM1Bfq3ySAd3bPSkWfAAgsoHVGGMQziPyp4XqjKKnrVEqTG2n9rekw1rMnJYOkvRlt(6zJho6Ip1iw9onQPcko6xIPcElnB6xG172GlpcjSEcfiLfWGoPADU6xp(47OjYoC5X9U8Op95p81387h7By0WULxIASEqwG5gcVJYRg1e009syeXGUD6b3PKdewVC1xEf927(R35)TqyA1yGP1eOPz5Puusmpvn9kwO8oUc75eB4QWwdy4wgv)S2ZRxxiwJ6UDZW130Z5WWRW5fqJkO1duIBgXYwh8ECxBKHCX4bdcqUMfygL41PBhpehI58eRNunLiOUF0QaQvjEZW41JVuXYaZFfDB72FiB0sAb30o1msgUoRKHkfljsPIewJcQoRJVHNdAfPxvf0Q4mSYmzNP3fsiA4BNFV5UwHIKoU0kLSlspAdOEwW7CcT749qv0J5uE7memYSUpH3yksy2H9jSEpHmVNFaR3R7)6toC8Q3dhV2Fq4rG5daIddG(pma6)Zba9pmao5HbWj)CaWHtb6gQc3eUvnwn)vPlbd03hTz4)9
```

With 2.16.6 deselecting then selecting a monster show (with custom code in conditions) "3+4+5" and "4" and "5"
With current master it does not

### PR DESCRIPTION
**Pull request by mrbuds**
# Description
Fixes #2015 #2017 #2020

### CODE REVIEW

**Comment by InfusOnWoW** [comment]
I pushed a different fix for the member case and fixed another bug.

**Comment by mrbuds** [comment]
New aura => status > cast > unit = specific unit
```
2x WeakAuras\Prototypes.lua:5658: attempt to concatenate local 'unit' (a nil value)
[string "@WeakAuras\Prototypes.lua"]:5658: in function `internal_events'
[string "@WeakAuras\GenericTrigger.lua"]:1210: in function `Add'
[string "@WeakAuras\WeakAuras-@project-version@.lua"]:4323: in function <WeakAuras\WeakAuras.lua:4270>
[string "@WeakAuras\WeakAuras-@project-version@.lua"]:4397: in function `Add'
[string "@WeakAurasOptions\WeakAurasOptions-@project-version@.lua"]:833: in function <WeakAurasOptions\WeakAurasOptions.lua:820>
[string "=[C]"]: ?
[string "@Ace3\AceConfig-3.0-3\AceConfigDialog-3.0\AceConfigDialog-3.0-78.lua"]:51: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:49>
[string "@Ace3\AceConfig-3.0-3\AceConfigDialog-3.0\AceConfigDialog-3.0-78.lua"]:843: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:664>
[string "=[C]"]: ?
[string "@Quartz\libs\AceGUI-3.0\AceGUI-3.0-40.lua"]:72: in function <Quartz\libs\AceGUI-3.0\AceGUI-3.0.lua:70>
[string "@Quartz\libs\AceGUI-3.0\AceGUI-3.0-40.lua"]:290: in function `Fire'
[string "@Ace3\AceGUI-3.0-40\widgets\AceGUIWidget-DropDown.lua"]:442: in function <...ns\Ace3\AceGUI-3.0\widgets\AceGUIWidget-DropDown.lua:433>
[string "=[C]"]: ?
[string "@Quartz\libs\AceGUI-3.0\AceGUI-3.0-40.lua"]:72: in function <Quartz\libs\AceGUI-3.0\AceGUI-3.0.lua:70>
[string "@Quartz\libs\AceGUI-3.0\AceGUI-3.0-40.lua"]:290: in function `Fire'
[string "@Ace3\AceGUI-3.0-40\widgets\AceGUIWidget-DropDown-Items.lua"]:351: in function <...3\AceGUI-3.0\widgets\AceGUIWidget-DropDown-Items.lua:341>
```

File: WeakAuras/Prototypes.lua [hunk 1]
```diff
@@ -1367,57 +1381,58 @@ local function AddUnitEventForEvents(result, unit, event)
...
+    local events = {}
+    if WeakAuras.multiUnitUnits[trigger.unit] then
+      for unit in pairs(WeakAuras.multiUnitUnits[trigger.unit]) do
+        tinsert(events, {"UNIT_CHANGED_" .. unit, unit})
+      end
+    else
+      if unit then
+        tinsert(events, {"UNIT_CHANGED_" .. unit, unit})
+      end
+    end
```
Comment (mrbuds) [inline WeakAuras/Prototypes.lua]: variable `unit` is overwritten in the for loop using same name
`unit` nil check in the else block should be done before the test of `WeakAuras.multiUnitUnits[trigger.unit]`

Comment (InfusOnWoW) [reply WeakAuras/Prototypes.lua]: The else block is never entered after the loop. But the code looks a bit convulted, I'll simplify it.

**Pull request merged**